File: Binary_Tree/226_invert_binary_tree/226.cpp
#include "../tree.cpp"
#include <vector>


BinaryTree *invertTree(BinaryTree *root) {
    if (root == nullptr) {
        return root;
    }
    BinaryTree* inverted_right = invertTree(root -> right);
    BinaryTree* inverted_left = invertTree(root -> left);
    root -> right = inverted_left;
    root -> left = inverted_right;
    return root; 
}


int main() {
    std::vector<int> array = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
    BinaryTree* root = build_tree(array);
    print_tree(root);
    std::cout << " " << std::endl;
    root = invertTree(root);
    print_tree(root);
}

File: Binary_Tree/226_invert_binary_tree/226.rs

#[path = "../tree.rs"]
mod tree;


fn invert_tree(root: Option<Box<tree::BinaryTree>>) -> Option<Box<tree::BinaryTree>> {

    match root {
        None => return None,
        Some(root) => {
            return tree::BinaryTree::new(root.val, invert_tree(root.right), invert_tree(root.left));
        }
    }
}


fn main() {
    let mut root = tree::build_tree(&mut vec!(0, 1, 2, 3, 4, 5, 6, 7, 8, 9));
    tree::print_tree(&root, 1);
    root = invert_tree(root);
    tree::print_tree(&root, 1);
}